### DISCUSSION TITLE
Endless/no break MP lobby setting?

### DISCUSSION THREAD

**Discussion by GOINGCRAZYWITHIT** [ppy/osu#24764] [Ideas] [answered]
I've been thinking about a setting that the host can turn on that does one of the following if not has all of them as options:

- ### A button is added at the score screen after a map that the host can hold to click to progress to the next map without any ready checks.

This option could be used in multiple ways or parted out and used as an extension of the lobby UI outside of this feature, allowing players to ready up at the score screen, maybe a separate button that goes back to the lobby AND readies them up, or maybe a toggle in the settings for auto-readying as soon as the map is over for those that wish to use it. Players can exit to the lobby at any point still.

- ### A countdown to the next map is shown to all players at the score screen after a map.

Edting of the duration of the countdown is optional as 5 - 10 seconds should be fine and waitng much longer would defeat the purpose of this option and likely be slower than a lobby that is on the ball and readying up quickly. Players can exit to the lobby during the countdown.

- ### A crossfade option that never takes any players out of the gameplay, as soon as one map ends, another begins, with audio and visual transitions.

I think this is the coolest out of the three. Imagine seeing a progress bar of the current map on lobbies with this option in use and being able to jump in whenever you want, obviously not getting a score for maps you joined in the middle of, but then smoothly transitioning into the next map as well as being able to sit out and join back in whenever you want. I feel that this is another option that could also be applied elsewhere, like to playlists, or a separate lobby setting that turns on joining/leaving of maps whenever, current map progress bars on all lobbies on hover. I almost feel like those last two one should be a default on setting as I don't see how they could negatively impact gameplay for anyone, but behind the scenes stuff of MP would need to be worked on first I imagine.

All of these would need to show an obvious badge or something to indicate that the setting is turned on and which option is chosen both in the lobby list and inside of the lobby.

I also feel showing existing UI elements in new places would benefit this new setting and it's possible options, like being able to see the current queue of maps during gameplay and possibly adding new maps through that new in-gameplay map list. In the case of the third option, this might be a little redundant for those not actively playing because they can just back out of the current map and add their own while they wait for the next, but in all other cases I think this would be a good addition. With that being said, I'm sure making this and the main topic a generally good user experience would be a difficult process.

**Comment by peppy** [comment, marked as answer]
Definitely something planned for the future!
